packages/jest-circus/src/utils.ts

```diff
@@ -264,12 +275,13 @@ export const callAsyncCircusFn = (
       // it's resolved.
       timeoutID.unref && timeoutID.unref();
       clearTimeout(timeoutID);
+      return setState({...getState(), seenDone: false});
```

Comment:
seems `setState` is not async, so just ditch the `await` from the previous commit, then

Reply:
I tried that, but the second `done` call was being made before the state was set, so I went ahead with async-await.

Reply:
Move your new logic out of `Promise.resolve().then(() => {` and do it above it

Reply:
Alright, sure!